src/view3d.cpp

```diff
@@ -271,23 +403,39 @@ void View3D::paintGL()
     // Draw bounding boxes
     if(m_drawBoundingBoxes && !m_incrementalDraw)
     {
-        for(size_t i = 0; i < geoms.size(); ++i)
-            drawBoundingBox(transState, geoms[i]->boundingBox(), Imath::C3f(1));
+        if (m_boundingBoxShader->isValid())
+        {
+            QGLShaderProgram &boundingBoxShader = m_boundingBoxShader->shaderProgram();
+            // shader
+            boundingBoxShader.bind();
+            // matrix stack
+            transState.setUniforms(boundingBoxShader.programId());
+
+            for (size_t i = 0; i < geoms.size(); ++i)
+            {
+                drawBoundingBox(transState, geoms[i]->getVAO("boundingbox"), geoms[i]->boundingBox().min,
+                                Imath::C3f(1), geoms[i]->shaderId("boundingbox")); //boundingBoxShader.programId()
+            }
+
+            // boundingBoxShader.release();
+        }
     }
 
     // Draw meshes and lines
     if (!m_incrementalDraw)
     {
-        drawMeshes(transState, geoms);
-
+        this->drawMeshes(transState, geoms);
```

Comment:
Purely as a matter of style consistency, could you remove the `this->` in various places?  It's not bad, just inconsistent with the rest of the code.